src/main/java/com/bb3/bodybuddybe/post/service/PostService.java

```diff
@@ -0,0 +1,84 @@
+package com.bb3.bodybuddybe.post.service;
+
+import com.bb3.bodybuddybe.common.security.UserDetailsImpl;
+import com.bb3.bodybuddybe.post.dto.PostCreateRequestDto;
+import com.bb3.bodybuddybe.post.dto.PostListResponseDto;
+import com.bb3.bodybuddybe.post.dto.PostResponseDto;
+import com.bb3.bodybuddybe.post.entity.Post;
+import com.bb3.bodybuddybe.post.repository.PostRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PostService {
+    private final PostRepository postRepository;
+
+    public void createPost(PostCreateRequestDto postCreateRequestDto, UserDetailsImpl userDetails) {
+
```

Comment:
요청 본문의 gymId로 Gym 객체를 찾아오고 존재하지 않는 경우 예외 처리하는 게 좋을 것 같습니다 ~~

```suggestion
Gym gym = gymRepository.findById(postCreateRequestDto.getGymId())
    .orElseThrow(() -> new CustomException(ErrorCode.GYM_NOT_FOUND));
```